### PR TITLE
fix: Date formatting and language translation for other regions

### LINKED ISSUE #7862: [CW-2457] fix: Make activity timestamp messages translatable
These copies aren't translatable at the moment. 

ref: [https://github.com/chatwoot/chatwoot/blob/9322112481e6341b6b291d7d44b234434a2f6c9b/app/javascript/dashboard/components/ui/TimeAgo.vue#L56](https://github.com/chatwoot/chatwoot/blob/9322112481e6341b6b291d7d44b234434a2f6c9b/app/javascript/dashboard/components/ui/TimeAgo.vue#L56)

<sub>From [SyncLinear.com](https://synclinear.com) | [CW-2457](https://linear.app/chatwoot/issue/CW-2457/fix-make-activity-timestamp-messages-translatable)</sub>

### PR DESCRIPTION
**Pull request by KokeroO**
# Pull Request Template

## Description

I decided to close PR https://github.com/chatwoot/chatwoot/pull/10360 and split it into 2 separate PRs.

1. When applying a new language setting from the account settings, date and time formatting is applied to the language of choice. Previously, only the 'en' localization was applied. This caused confusion and doubt for the end user.
2. Other changes have also been applied to translate texts that were not handled by the i18n library ​​that contain region code.

Resolves https://github.com/chatwoot/chatwoot/issues/7862, https://github.com/chatwoot/chatwoot/issues/9285, https://github.com/chatwoot/chatwoot/issues/10327

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## How Has This Been Tested?

1. Go to app.chatwoot.com/app/accounts/1/settings/general and change the account language.
2. All pages showing dates will be in the correct format for your locale.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by KokeroO** [comment]
@scmmishra 
https://github.com/chatwoot/chatwoot/pull/10360#pullrequestreview-2403714402

1. Fixed
2. It is fixed
3. I couldn't bring it to the vue context, the only better way I found was this. Maybe if you implement the script within the vue context.
4. Datepicker has the problem that it does not use the language chosen by the user. It would be possible to use `window` to set values ​​from within Vue, but in the helper we would have to use `setInterval()` to update the value inside the helper, since the value does not update itself.
It would be possible to use `MutationObserver()`, but this wastes unnecessary resources to update a single value instead of looking at the changes in the entire DOM.


In short it is now working.

I thank.

**Comment by scmmishra** [comment]
Hey @KokeroO this looks great

> * I couldn't bring it to the vue context, the only better way I found was this. Maybe if you implement the script within the vue context.

For this perhaps, you could make file the following file `app/javascript/dashboard/composables/useLocaleFormatter.js`

```js
import { useI18n } from 'vue-i18n';
import {
  messageStamp,
  messageTimestamp,
  dynamicTime,
  dateFormat,
  shortTimestamp,
} from 'shared/helpers/timeHelper';

export default function useLocaleFormatter() {
  const { locale } = useI18n();

  const localeDynamicTime = (...args) => dynamicTime(...args, locale.value);
  const localeDateFormat = (...args) => dateFormat(...args, locale.value);
  const localeShortTimestamp = (...args) =>
    shortTimestamp(...args, locale.value);
  const localeMessageStamp = (...args) => messageStamp(...args, locale.value);
  const localeMessageTimestamp = (...args) =>
    messageTimestamp(...args, locale.value);

  return {
    localeDynamicTime,
    localeDateFormat,
    localeShortTimestamp,
    localeMessageStamp,
    localeMessageTimestamp,
  };
}
```


This can be used anywhere in the setup context as 

```
const { localeDateFormat } = useLocaleFormatter();
```

**Comment by KokeroO** [comment]
@pranavrajs
I have some doubts regarding `dateFormat`, `messageStamp` and `messageTimestamp`.
They confuse me, originally in certain situations they give the same result.
Could you take a look? I could unify them.

I just need to fix timeHelper.js and timeHelper.spec.js.

app/javascript/shared/helpers/specs/timeHelper.spec.js
app/javascript/shared/helpers/timeHelper.js

**Comment by scmmishra** [comment]
> Could you take a look? I could unify them.

Yeah, seems like they can be combined. Just need to handle the `isSameYear` case in `messageTimestamp`. But we can combine `dateFormat` and `messageStamp` for sure

**Comment by KokeroO** [comment]
> > Could you take a look? I could unify them.
> 
> Yeah, seems like they can be combined. Just need to handle the `isSameYear` case in `messageTimestamp`. But we can combine `dateFormat` and `messageStamp` for sure

@scmmishra 
Thanks for the tips. I was able to implement it and do the best I could. This PR will greatly help other languages ​​and I believe that the dates are now better represented.

**Comment by scmmishra** [comment]
@KokeroO this looks great! I'll review this ASAP and let you know

**Comment by Micorksen** [comment]
Hey! I see that my issue was mentionned here.
In fact, my issue is about the help center and wasn't really about the agent UI...

**Comment by KokeroO** [comment]
@Micorksen The help center will also be covered in this PR.

**Comment by Micorksen** [comment]
Hey @pranavrajs @sojan-official, can someone please accept this PR?

**Comment by sojan-official** [comment]
@sony-mathew @iamsivin  can we look into getting this for https://github.com/chatwoot/chatwoot/milestone/121

**Comment by sony-mathew** [comment]
@copilot resolve the merge conflicts in this pull request
